src/util/ThemeCache.js

```diff
@@ -30,37 +30,40 @@ const getThemeItemTimestamps = async (url: string): Promise<ItemTimestamps> => {
   return { start: Date.parse(start), expiration: Date.parse(expiration) }
 }
 
-export async function getBackgroundImageFromCDN(disklet: Disklet): Promise<ImageSourcePropType> {
+const downloadFile = async (disklet: Disklet, fromUrl: string, toFile: string): Promise<void> => {
+  try {
+    // See if the server even has an image for us in the first place:
+    const cache: ThemeCache = await getThemeCache(disklet).catch(() => ({}))
+    const { start, expiration } = await getThemeItemTimestamps(fromUrl)
+    if (cache[fromUrl] != null && cache[fromUrl].start === start && cache[fromUrl].expiration === expiration) return
+    // Download file whenever local file timestamps do not equal remote file timestamps
+    const download = RNFS.downloadFile({ fromUrl, toFile })
+    const status = await download.promise
+    if (status.statusCode !== 200) throw new Error('Failed to download')
+    cache[fromUrl] = { start, expiration }
+    await setThemeCache(disklet, cache)
+  } catch (e) {
+    console.warn(`Error downloading ${fromUrl}`)
+  }
+}
+
+export async function getBackgroundImage(disklet: Disklet, fallback: ImageSourcePropType): Promise<ImageSourcePropType> {
   const BACKGROUND_IMAGE_FILE_NAME = 'login_bg.gif'
   const BACKGROUND_IMAGE_URL = `${EDGE_CONTENT_SERVER}/${BACKGROUND_IMAGE_FILE_NAME}`
   const BACKGROUND_IMAGE_LOCAL_URI = `file://${directory}/${BACKGROUND_IMAGE_FILE_NAME}`
   const now = Date.now()
+  let image = fallback
 
   const cache: ThemeCache = await getThemeCache(disklet).catch(() => ({}))
-  if (cache[BACKGROUND_IMAGE_URL] == null) {
-    cache[BACKGROUND_IMAGE_URL] = { cachedTimestamp: now }
-  }
-
-  try {
-    const timestamps = await getThemeItemTimestamps(BACKGROUND_IMAGE_URL)
-    if (now > timestamps.start && now < timestamps.expiration) {
-      if (
-        cache[BACKGROUND_IMAGE_URL].cachedTimestamp > timestamps.start &&
-        cache[BACKGROUND_IMAGE_URL].cachedTimestamp < timestamps.expiration &&
-        (await RNFS.exists(BACKGROUND_IMAGE_LOCAL_URI))
-      ) {
-        return { uri: BACKGROUND_IMAGE_LOCAL_URI }
-      }
-
-      await RNFS.downloadFile({ fromUrl: BACKGROUND_IMAGE_URL, toFile: BACKGROUND_IMAGE_LOCAL_URI })
-      cache[BACKGROUND_IMAGE_URL].cachedTimestamp = now
-      await setThemeCache(disklet, cache)
-      return { uri: BACKGROUND_IMAGE_LOCAL_URI }
-    } else {
-      throw new Error('CDN file expired.')
-    }
-  } catch (e) {
-    console.warn(`Error accessing CDN at ${BACKGROUND_IMAGE_URL} or local cache`)
-    throw e
+  if (
+    cache[BACKGROUND_IMAGE_URL] != null &&
+    cache[BACKGROUND_IMAGE_URL].start < now &&
+    cache[BACKGROUND_IMAGE_URL].expiration > now &&
+    (await RNFS.exists(BACKGROUND_IMAGE_LOCAL_URI))
+  ) {
+    image = { uri: BACKGROUND_IMAGE_LOCAL_URI }
   }
+  // Always return existing local file but query and download new remote file in the background
+  downloadFile(disklet, BACKGROUND_IMAGE_URL, BACKGROUND_IMAGE_LOCAL_URI)
```

Comment:
Need to catch if downloadFile throws or the app might crash

Reply:
`downloadFile(disklet, BACKGROUND_IMAGE_URL, BACKGROUND_IMAGE_LOCAL_URI).catch(() => {})`